source/uwp/Renderer/lib/Util.cpp

```diff
@@ -1312,18 +1312,14 @@ HRESULT ProjectedElementTypeToHString(ABI::AdaptiveNamespace::ElementType projec
 
 HRESULT IsBackgroundImageValid(_In_ ABI::AdaptiveNamespace::IAdaptiveBackgroundImage* backgroundImageElement, _Out_ BOOL* isValid)
 {
+    *isValid = FALSE;
     ComPtr<ABI::AdaptiveNamespace::IAdaptiveBackgroundImage> backgroundImage(backgroundImageElement);
     if (backgroundImage != NULL)
     {
-        HSTRING url;
-        THROW_IF_FAILED(backgroundImage->get_Url(&url));
-        if (url != NULL)
-        {
-            *isValid = TRUE;
-            return S_OK;
-        }
+        HString url;
+        THROW_IF_FAILED(backgroundImage->get_Url(url.GetAddressOf()));
```

Comment:
>THROW_IF_FAILED [](start = 8, length = 15)

hmm we are mixing exceptions wtih suposedly error contracts here.. RETURN_IF_FAILED()

Reply:
in general we have a mix because a lot of these helpers were originally written as throwing. however, it became apparent that in some scenarios exceptions weren't sufficient (IIRC, there was a problem where thrown exceptions wouldn't cross our ABI boundaries correctly or something). #2534 covers reconciling our error handling regimes. that said, I did find at least one place where an exception from here wouldn't get properly handled, so I'll make this `RETURN_IF_FAILED()`.

---
In reply to: [316950627](https://github.com/microsoft/AdaptiveCards/pull/3380#discussion_r316950627) [](ancestors = 316950627)